themes-default/slim/views/viewlogs.mako

```diff
@@ -15,43 +30,86 @@ const startVue = () => {
+                    levels = LOGGING_LEVELS.keys()
+                    levels.sort(lambda x, y: cmp(LOGGING_LEVELS[x], LOGGING_LEVELS[y]))
+                    if not app.DEBUG:
+                        levels.remove('DEBUG')
+                    if not app.DBDEBUG:
+                        levels.remove('DB')
+                %>
+                LoggingLevelLabels: JSON.parse('${json.dumps(LOGGING_LEVELS)}'),
+                loggingLevels: JSON.parse('${json.dumps(levels)}')
             };
         },
-        mounted() {
-            function getParams() {
+        computed: {
+            params() {
+                const { log } = this;
+                const { minLevel, filter, period, search } = log;
+
                 return $.param({
-                    min_level: $('select[name=min_level]').val(), // eslint-disable-line camelcase
-                    log_filter: $('select[name=log_filter]').val(), // eslint-disable-line camelcase
-                    log_period: $('select[name=log_period]').val(), // eslint-disable-line camelcase
-                    log_search: $('#log_search').val() // eslint-disable-line camelcase
+                    min_level: minLevel,
+                    log_filter: filter,
+                    log_period: period,
+                    log_search: search
                 });
             }
+        },
+        methods: {
+            viewLogAsText() {
+                const { params } = this;
 
-            $('#min_level,#log_filter,#log_search,#log_period').on('keyup change', _.debounce(() => {
-                const params = getParams();
-                $('#min_level').prop('disabled', true);
-                $('#log_filter').prop('disabled', true);
-                $('#log_period').prop('disabled', true);
-                document.body.style.cursor = 'wait';
-
-                $.get('errorlogs/viewlog/?' + params, data => {
-                    history.pushState('data', '', 'errorlogs/viewlog/?' + params);
-                    $('pre').html($(data).find('pre').html());
-                    $('#min_level').prop('disabled', false);
-                    $('#log_filter').prop('disabled', false);
-                    $('#log_period').prop('disabled', false);
-                    document.body.style.cursor = 'default';
-                });
-            }, 500));
-
-            $(document.body).on('click', '#viewlog-text-view', e => {
-                e.preventDefault();
-                const params = getParams();
                 const win = window.open('errorlogs/viewlog/?' + params + '&text_view=1', '_blank');
                 win.focus();
-            });
+            },
+            async getLogs() {
+                const { params } = this;
+
+                this.disabled = true;
+
+                const data = await $.get('errorlogs/viewlog/?' + params);
+                history.pushState('data', '', 'errorlogs/viewlog/?' + params);
+
+                this.logs = $(data).find('pre').html();
```

Comment:
This makes https://github.com/pymedusa/Medusa/pull/4104/commits/df69d36f2e2228781a66c1d25304ee0d88e9b133 unusable.